lib/inc/sai_redis_internal.h

```diff
@@ -184,7 +184,7 @@
     sai_status_t redis_clear_ ## object_type ## _stats(             \
             _In_ sai_object_id_t object_type ## _id,                \
             _In_ uint32_t number_of_counters,                       \
-            _In_ const sai_ ## object_type ## _stat_t *counter_ids) \
+            _In_ const sai_stat_id_t *counter_ids) \
```

Comment:
i preffered to have separate type for each stats, this gave extra check for type, so user would not pass different stats for given objects at compile time
is there any reason you are changing this ?

Reply:
I'm afraid it's not possible to keep the old type, because this function then cannot be casted to a function pointer of the SAI API, that is using sai_stat_id_t.

Reply:
yes, each new function pointer needs to be created for each stat, and this is how it was so far

Reply:
Now in SAI stat functions look like this:

```
typedef sai_status_t (*sai_clear_bfd_session_stats_fn)(
        _In_ sai_object_id_t bfd_session_id,
        _In_ uint32_t number_of_counters,
        _In_ const **sai_stat_id_t** *counter_ids);
```

So if I leave macros as is, it will fail to compile with following error:

`error: invalid conversion from 'sai_status_t (*)(sai_object_id_t, uint32_t, const sai_bfd_session_stat_t*) {aka int (*)(long unsigned int, unsigned int, const _sai_bfd_session_stat_t*)}' to 'sai_clear_bfd_session_stats_fn {aka int (*)(long unsigned int, unsigned int, const unsigned int*)}' [-fpermissive]`

Maybe I'm missing your point, but doesn't redis SAI API have to look exactly like SAI api?
These changes to sairedis are done with an assumption that SAI API is taken as is.